From 55f8f524d1081ff2679fa4e5424af21dddf40eeb Mon Sep 17 00:00:00 2001
From: Philip Cuadra <philipcuadra@google.com>
Date: Wed, 5 Jul 2017 21:08:22 -0700
Subject: [PATCH 072/420] tty:  check before stopping kthread

Kthread allocation can fail, so check that it's not an error value
before trying to stop it.

Bug: 63354008
Test: build & run bluetooth audio
Signed-off-by: Philip Cuadra <philipcuadra@google.com>

Change-Id: Ia8a91645beef2b4df64582b9059272f6df8ad4a9
---
 drivers/tty/tty_buffer.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/tty/tty_buffer.c b/drivers/tty/tty_buffer.c
index 8917b6f87b04..92af201f9030 100644
--- a/drivers/tty/tty_buffer.c
+++ b/drivers/tty/tty_buffer.c
@@ -132,7 +132,8 @@ void tty_buffer_free_all(struct tty_port *port)
 	buf->tail = &buf->sentinel;
 
 	atomic_set(&buf->mem_used, 0);
-	kthread_stop(port->worker_thread);
+	if (!IS_ERR_OR_NULL(port->worker_thread))
+		kthread_stop(port->worker_thread);
 }
 
 /**
-- 
2.15.0

